[material-ui][docs] Revise the Alert demo page

samuelsycamore opened the pull request:
Part of #35158 

https://deploy-preview-34892--material-ui.netlify.app/material-ui/react-alert/

oliviertassinari commented on docs/data/material/components/alert/alert.md:
````diff
@@ -8,81 +8,155 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
 
-### Transition
+```html
+<div
+  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard"
+  role="alert"
+>
````
**Off-topic**

I don't recall what's the current state of the reflection on the non-slot class names (I don't know if there is a clear GitHub issue about it). If the direction is to remove non-slot class names in v6, then it will like this:

```suggestion
<div class="MuiPaper-root MuiAlert-root" role="alert">
```

It would make the DOM feel lighter, at the expense of making it harder to customize the component with CSS Modules or SAAS like it's currently possible with Bootstrap https://getbootstrap.com/docs/5.2/components/alerts/ or Vuetify. https://vuetifyjs.com/en/components/alerts/. These people would need to use the `classes` API.

Another option is https://twitter.com/devongovett/status/1576635415024390144

oliviertassinari commented on docs/data/material/components/alert/alert.md:
```diff
@@ -8,81 +8,155 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
+## Usage
 
-The alert offers four severity levels that set a distinctive icon and color.
+After [installation](/material/getting-started/installation/), you can start building with this component using the following basic elements:
```
```suggestion
After [installation](/material-ui/getting-started/installation/), you can start building with this component using the following basic elements:
```

oliviertassinari commented on docs/data/material/components/alert/alert.md:
```diff
@@ -8,81 +8,155 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
 {{"demo": "ColorAlerts.js"}}
 
-## Accessibility
+### Transitions
```
I'm not sure it's customization. I mean, where does customization ends? Since "Overriding the root slot" is 't part of it, I would imagine it's about visual design

```suggestion
## Transitions
```

samuelsycamore replied:
Good point about the Transitions section. As for "Overriding the root slot," I'm torn about that decision. I feel that the info is more closely related to the "Anatomy" section, but it would also make sense under "Customization."

samuelsycamore replied:
@oliviertassinari another option would be to move the "Overriding" info further down the page, maybe to a new section called "Advanced" below "Customization"? Since I assume the majority of users are not going to bother with it.

oliviertassinari replied:
> to move the "Overriding" info further down the page, maybe to a new section called "Advanced" below "Customization"

I would remove this `Overriding the root slot` section altogether. It sounds like something all the components support, so not worth to be duplicate on each component guide page.

samuelsycamore replied:
So then, don't mention this info at all on the Material UI component pages?

oliviertassinari replied:
Yeah, and maybe only cover this in a customization guide 🤷‍♂️

samuelsycamore replied:
Hmm, well then before I move forward with more component page revisions like this one, I'll write that customization guide and then we can decide how it should connect with these pages.

oliviertassinari replied:
For example in https://mui.com/base/react-select/#slot-props, we are describing the API of the `SelectUnstyled` component. It feels like something that would be better in the reference docs: https://mui.com/base/api/select-unstyled/.

oliviertassinari commented on docs/data/material/components/alert/alert.md:
```diff
@@ -8,81 +8,155 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
-Actions must have a tab index of 0 so that they can be reached by keyboard-only users.
+- Because alerts are not intended to interfere with the use of the app, your Alert component should _never_ affect the keyboard focus.
+- If an alert contains an action, that action must have a `tabindex` of `0` so it can be reached by keyboard-only users.
+- Alerts should not disappear automatically—[timed interactions](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-no-exceptions.html) can make your app inaccessible to users who need extra time to understand or locate the alert.
```
```suggestion
- Essential Alerts should not disappear automatically—[timed interactions](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-no-exceptions.html) can make your app inaccessible to users who need extra time to understand or locate the alert.
```

samuelsycamore commented:
@oliviertassinari I removed the info about the overriding props, and moved the Anatomy section to the bottom of the page. We discussed creating a new tab on the Component page for Anatomy in a Core meeting, but until we implement it, I think this page structure works. What do you think?

oliviertassinari commented:
@samuelsycamore Cool, based on https://deploy-preview-34892--material-ui.netlify.app/material-ui/react-alert/, I think going in the same direction as https://github.com/mui/material-ui/pull/35152#issuecomment-1315886747 could be valuable:

1. We could remove `## Usage` with the argument that it duplicates with `## Basics` and that we don't need installation instructions. 
2. We could make the `## Basics` demos really basic, the demo we have sounds more like a demo of the severity prop. We could then make the demo code expanded by default with `, "defaultCodeOpen": true`.
3. For the anatomy section, I think that it could be higher in the tree as it's meant to help understand how the component is structured.

samuelsycamore commented:
I'm putting this PR on hold until we finalize the content structure for the Base and Joy docs—then we should be able to circle back and apply consistent formatting to the Material UI component docs like this one.

danilo-leal commented on docs/data/material/components/alert/alert.md:
```diff
@@ -8,83 +8,127 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
+Use the Material UI [Dialog](https://mui.com/material-ui/react-dialog/) if you need this behavior.
 :::
 
 {{"component": "modules/components/ComponentLinkHeader.js"}}
```
I feel like the best place to position this component here is right below the description (or, effectively, below the ad container). What do you think? It's how we did in the recent Divider and Accordion pages. It looks like a cohesive set of very introductory links — particularly on the Base UI docs, where the ad container is below the tabs (a layout we should get to, eventually!).

danilo-leal commented on docs/data/material/components/alert/alert.md:
```diff
@@ -8,83 +8,127 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
+## Introduction
+
+The Alert component can be used to provide important and potentially time-sensitive information in a way that does not interfere with the user's tasks. (Source: [ARIA APG](https://www.w3.org/WAI/ARIA/apg/patterns/alert/).)
+
```
Maybe a very, very simple demo here? Like the one we recently did on the Accordion page.

danilo-leal commented on docs/data/material/components/alert/alert.md:
```diff
@@ -8,83 +8,127 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/alert/
+
+The Alert component can be used to provide important and potentially time-sensitive information in a way that does not interfere with the user's tasks. (Source: [ARIA APG](https://www.w3.org/WAI/ARIA/apg/patterns/alert/).)
+
+This component is not documented in the [Material Design guidelines](https://material.io/).
```
We used an info callout on the Accordion page for this info — should we standardize it?

danilo-leal reviewed (approved):
Looks awesome — dropping an approval already! Left some somewhat small considerations there.

danilo-leal reviewed (approved):
Looks awesome! 🎉
